File: lib/models/login_response.dart
class LoginResponse{
    String? username;
    String? accessToken;
    String? role;
    LoginResponse({this.username, this.role, this.accessToken});

    factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(
      username: json['username'],
      role: json['role'],
      accessToken: json['accessToken']
    );
    }
}

File: back-end/src/main/java/com/groupproject/softwaretu/apiv1/profile/LoginResponse.java
package com.groupproject.softwaretu.apiv1.profile;

import lombok.Data;
import com.groupproject.softwaretu.security.User;

@Data
public class LoginResponse {
    private String username;
    private String role;
    private String accessToken;
    public LoginResponse(User user){
        this.username = user.getUsername();
        this.role = user.getRole();
    }
}